crates/dpapi-core/Cargo.toml

```diff
@@ -0,0 +1,20 @@
+[package]
+name = "dpapi-core"
+version = "0.1.0"
+edition = "2024"
+
+[features]
+std = []
+
+[dependencies]
+bitflags.workspace = true
+num-derive.workspace = true
+num-traits = { workspace = true, default-features = true }
```

Comment:
note: I typically recommend against using `num-derive` and `num-traits` crates.

Two main reasons:

- [For resilient parsing, we should typically avoid enums](https://github.com/Devolutions/IronRDP/tree/b72e0857bfe2bcf502db62e08da605412d9ba80c/crates/ironrdp-pdu#enumeration-like-types-should-allow-resilient-parsing), and
- Otherwise, [it’s typically not worth adding a proc-macro in a core crate](https://github.com/Devolutions/IronRDP/blob/b72e0857bfe2bcf502db62e08da605412d9ba80c/ARCHITECTURE.md#core-tier). (This also applies to `thiserror`.)

Reply:
Note that I still recommend against using `num-derive` and `num-traits` crates even in `dpapi-pdu`. But if you are sure it covers your needs properly, and if it’s really saving you trouble, it’s fine!